C/tasks/purge_system/purge_system.cpp

```diff
@@ -0,0 +1,168 @@
+/*
+ * Fledge Statistics History
+ *
+ * Copyright (c) 2021 Dianomic Systems
+ *
+ * Released under the Apache 2.0 Licence
+ *
+ * Author: Stefano Simonelli
+ */
+
+#include <purge_system.h>
+#include <logger.h>
+
+#include <cstdarg>     /* va_list, va_start, va_arg, va_end */
+#include <cstdlib>
+#include <thread>
+#include <csignal>
+
+using namespace std;
+
+volatile std::sig_atomic_t signalReceived = 0;
+
+static const string DEFAULT_CONFIG = QUOTE({
+	"retainStatsHistory": {
+		"description": "This is the measure of how long to retain statistics history data for and should be measured in days.",
+		"type": "integer",
+		"default": "7",
+		"displayName": "Retain Stats History Data (In Days)",
+		"order": "1",
+		"minimum": "1"
+	},
+	"retainAuditLog" : {
+		"description": "This is the measure of how long to retain audit trail information for and should be measured in days.",
```

Comment:
"The number of days for which audit trail data is retained"

Reply:
done